src/test/scala-3/com/raquo/airstream/split/SplittableTypeSpec.scala

```diff
@@ -0,0 +1,474 @@
+package com.raquo.airstream.split
+
+import com.raquo.airstream.UnitSpec
+import com.raquo.airstream.eventbus.EventBus
+import com.raquo.airstream.fixtures.{Effect, TestableOwner}
+import com.raquo.airstream.state.Var
+import com.raquo.airstream.split.SplittableTypeMacros.{splitMatch, handleCase, toSignal, toStream}
+
+import scala.collection.{immutable, mutable}
+import scala.scalajs.js
+import com.raquo.airstream.ShouldSyntax.shouldBeEmpty
+
+class SplittableTypeSpec extends UnitSpec {
+
+  sealed trait Foo
+
+  final case class Bar(strOpt: Option[String]) extends Foo
+  enum Baz extends Foo {
+    case Baz1, Baz2
+  }
+  case object Tar extends Foo
+
+  final case class Res(result: Any)
+  
+  it("split match signal") {
+    val effects = mutable.Buffer[Effect[String]]()
+
+    val myVar = Var[Foo](Bar(Some("initial")))
+
+    val owner = new TestableOwner
+
+    val signal = myVar.signal
+      .splitMatch
+      .handleCase {
+        case Bar(Some(str)) => str
+        case Bar(None) => "null"
+      } { case (str, strSignal) =>
+        effects += Effect("init-child", s"Bar-$str")
+        // @Note keep foreach or addObserver here – this is important.
+        //  It tests that SplitSignal does not cause an infinite loop trying to evaluate its initialValue.
+        strSignal.foreach { str =>
+          effects += Effect("update-child", s"Bar-$str")
+        }(owner)
+
+        Res(str)
+      }
+      .handleCase { case baz: Baz => baz } { case (baz, bazSignal) =>
+        effects += Effect("init-child", s"Baz-${baz.ordinal}-${baz.toString}")
+        // @Note keep foreach or addObserver here – this is important.
+        //  It tests that SplitSignal does not cause an infinite loop trying to evaluate its initialValue.
+        bazSignal.foreach { baz =>
+          effects += Effect("update-child", s"Baz-${baz.ordinal}-${baz.toString}")
+        }(owner)
+
+        Res(baz)
+      }
+      .handleCase { case Tar => 10 } { case (int, intSignal) =>
+        effects += Effect("init-child", s"Tar-${int}")
+        // @Note keep foreach or addObserver here – this is important.
+        //  It tests that SplitSignal does not cause an infinite loop trying to evaluate its initialValue.
+        intSignal.foreach { int =>
+          effects += Effect("update-child", s"Tar-${int}")
+        }(owner)
+
+        Res(int)
+      }
+      .toSignal
+    
+    signal.foreach { result =>
+      effects += Effect("result", result.toString)
+    }(owner)
+
+    effects shouldBe mutable.Buffer(
+      Effect("init-child", "Bar-initial"),
+      Effect("update-child", "Bar-initial"),
+      Effect("result", "Res(initial)")
+    )
+
+    effects.clear()
+
+    myVar.writer.onNext(Bar(None))
+
+    effects shouldBe mutable.Buffer(
+      Effect("result", "Res(initial)"),
+      Effect("update-child", "Bar-null")
+    )
```

Comment:
I was really confused by these assertions. Intuitively, it does not make sense to expect the result to be "initial" when you have emitted "null" – at this point, "initial" is a stale value already.

I now see the root of my confusion – you're returning `Res(<initialValue>)` from the split callbacks, such as `Res(str)` or `Res(baz)`. Whereas my other tests return the `key` that the callback is for, e.g. see the callbacks returning `Bar(key)` in `SplitsignalSpec` – the resulting assertions are more intuitive there.

To help me keep my sanity, can you please replace usages of `Res(<initialvalue>)` with `Res(<key>)`, for example `Res("Bar")` instead of `Res(str)`, and `Res("Baz")` instead of `Res(baz)` above – throughout all of these new tests.

The real `key`-s used by the macros are hidden, and are integers, so you should make up semantic string keys that would basically act as debug labels to show which of the `handleCase`-s the callback belongs to, like "Bar" and "Baz" in my examples above.

Reply:
Fair enough, I'll changes these test